tests/integration-tests/clusters_factory.py

```diff
@@ -63,34 +63,28 @@ def cfn_outputs(self):
 class ClustersFactory:
     """Manage creation and destruction of pcluster clusters."""
 
-    def __init__(self, ssh_key, test_name=None):
-        self.__test_name = test_name
-        self.__ssh_key = ssh_key
+    def __init__(self):
         self.__created_clusters = {}
 
-    def create_cluster(self, config, name=None):
+    def create_cluster(self, cluster):
```

Comment:
It is a little weird to pass the _cluster_ to create to the `create_cluster` method.

What about a `cluster_definition` (`ClusterDef`)?
The same applies for the `create_stack`.

Alternatively we could rename the `create_cluster/stack` to `instantiate_cluster/stack`

Reply:
as agreed we'll leave this refactoring for a later PR